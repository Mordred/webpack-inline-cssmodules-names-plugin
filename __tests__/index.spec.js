/**
 * External dependencies
 */
const fs = require('fs');
const path = require('path');
const rimraf = require('rimraf');
const webpack = require('webpack');
const InlineCSSModulesNamesPlugin = require('..');

describe('webpack-inline-cssmodules-names-plugin', () => {
	const fixturesDirectory = path.join(__dirname, 'fixtures');
	const buildDirectory = path.join(__dirname, 'build');

	beforeAll(() => {
		rimraf.sync(buildDirectory);
	});

	afterAll(() => {
		rimraf.sync(buildDirectory);
	});

	test('should remove constant classname keys', done => {
		const inputDirectory = path.join(fixturesDirectory, 'basic');
		const outputDirectory = path.join(buildDirectory, 'basic');
		const config = {
			context: inputDirectory,
			entry: './index.js',
			mode: 'production',
			target: 'node',
			module: {
				rules: [
					{
						test: /\.css/,
						use: [{
							loader: 'css-loader',
							options: {
								modules: true,
								onlyLocals: true,
							},
						}]
					}
				]
			},
			optimization: {
				runtimeChunk: false,
				moduleIds: 'named',
				chunkIds: 'named',
				minimize: false,
			},
			output: {
				path: outputDirectory,
			},
			plugins: [
				new InlineCSSModulesNamesPlugin([/\.css$/]),
			],
		};

		webpack(config, err => {
			expect(err).toBeNull();

			const outputFile = path.join(outputDirectory, 'main.js');
			const outputFileContent = fs.readFileSync(outputFile, 'utf8');
			expect(outputFileContent).toMatchSnapshot('Output bundle should match snapshot');

			done();
		});
	});

	test('should produce expected output', done => {
		const inputDirectory = path.join(fixturesDirectory, 'with-variable');
		const outputDirectory = path.join(buildDirectory, 'with-variable');
		const config = {
			context: inputDirectory,
			entry: './index.js',
			mode: 'production',
			target: 'node',
			module: {
				rules: [
					{
						test: /\.css/,
						use: [{
							loader: 'css-loader',
							options: {
								modules: true,
								onlyLocals: true,
							},
						}]
					}
				]
			},
			optimization: {
				runtimeChunk: false,
				moduleIds: 'named',
				chunkIds: 'named',
				minimize: false,
			},
			output: {
				path: outputDirectory,
			},
			plugins: [
				new InlineCSSModulesNamesPlugin([/\.css$/]),
			],
		};

		webpack(config, err => {
			expect(err).toBeNull();

			const outputFile = path.join(outputDirectory, 'main.js');
			const outputFileContent = fs.readFileSync(outputFile, 'utf8');
			expect(outputFileContent).toMatchSnapshot('Output bundle should match snapshot');

			done();
		});
	});

});