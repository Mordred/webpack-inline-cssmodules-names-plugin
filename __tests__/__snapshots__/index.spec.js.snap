// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-inline-cssmodules-names-plugin should produce expected output: Output bundle should match snapshot 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./index.js\\":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);

// EXTERNAL MODULE: ./styles.css
var styles = __webpack_require__(\\"./styles.css\\");
var styles_default = /*#__PURE__*/__webpack_require__.n(styles);

// CONCATENATED MODULE: ./utils.js
function util() {
    return 'util';
}
// CONCATENATED MODULE: ./index.js



console.log(/* inline-classname */ \\"_1pnf1zespOZ6Wh_39JYiwU\\");
console.log(/* inline-classname */ \\"IOSoqXoyZpErJ5EF7paGS\\");
const mode = 'variable';
console.log(styles_default.a[\`Example__button--\${mode}\`]);
console.log(util());


/***/ }),

/***/ \\"./styles.css\\":
/***/ (function(module, exports) {

// Exports
module.exports = {
	\\"Example\\": \\"_1pnf1zespOZ6Wh_39JYiwU\\",
	\\"Example__button\\": \\"IOSoqXoyZpErJ5EF7paGS\\",
	\\"Example__button--variable\\": \\"_2yOeyO_4TJVeZD75xvzd65\\"
};

/***/ })

/******/ });"
`;

exports[`webpack-inline-cssmodules-names-plugin should remove constant classname keys: Output bundle should match snapshot 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./index.js\\":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);

// CONCATENATED MODULE: ./utils.js
function util() {
    return 'util';
}
// CONCATENATED MODULE: ./index.js



console.log(/* inline-classname */ \\"_1pnf1zespOZ6Wh_39JYiwU\\");
console.log(/* inline-classname */ \\"IOSoqXoyZpErJ5EF7paGS\\");
console.log(/* inline-classname */ \\"_1qQUo422N-U7RCEqM4aqto\\");
console.log(/* inline-classname */ \\"_3b010Cfq7ukTkIzod4hPKN\\");
console.log(/* inline-classname */ \\"UtJJNh9g54kBLKG0qWQaI\\");
console.log(util());

/***/ })

/******/ });"
`;
